cli/internal/run/global_hash.go

```diff
@@ -46,44 +68,52 @@ type oldGlobalHashable struct {
 	pipeline             fs.PristinePipeline
 }
 
-// calculateGlobalHashFromHashable returns a hash string from the globalHashable
-func calculateGlobalHashFromHashable(full GlobalHashable) (string, error) {
+// oldGlobalHash is a transformation of GlobalHashableInputs.
+// This exists because the existing global hashes are still usable
+// in some configurations that do not include a specified `EnvMode`.
+// We can remove this whenever we want to migrate users.
```

Comment:
It might be useful to add a searchable sigil like "TODO: Invalidate global hash"  or whatever that's searrchabl. We're tracking this already in linear, but maybe in the codebase would be more useful? (not a blocker)

Reply:
The PR for this is already up where we drop the `pipeline`.